assets/src/edit-story/components/checklistCard/stories/index.js

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { action } from '@storybook/addon-actions';
+import styled from 'styled-components';
+/**
+ * Internal dependencies
+ */
+import { ChecklistCard } from '..';
+import {
+  Button,
+  BUTTON_SIZES,
+  BUTTON_TYPES,
+  THEME_CONSTANTS,
+  Text,
+  Link,
+  List,
+} from '../../../../design-system';
+import { Thumbnail, THUMBNAIL_TYPES } from '../../thumbnail';
+import { THUMBNAIL_BG } from '../../thumbnail/stories/demoThumbnails';
+import { CARD_TYPE } from '../constants';
+import { CardListWrapper } from '../styles';
+
+export default {
+  title: 'Stories Editor/Components/ChecklistCard',
+  component: ChecklistCard,
+};
+
+const Container = styled.div`
+  background-color: ${({ theme }) => theme.colors.bg.primary};
+  display: flex;
+  flex-wrap: wrap;
+  padding: 16px;
+  width: 900px;
+
+  & > div {
+    margin: 0 8px 16px;
+  }
+`;
+
+export const _default = () => {
+  return (
+    <Container>
+      <div>
+        <Text>{'Single Issue'}</Text>
+        <ChecklistCard
+          title="Add video captions"
+          titleProps={{
+            onClick: () => action('title clicked')(),
+          }}
+          footer={
+            <Text size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.X_SMALL}>
+              {
+                'Keep the audience engaged even when they can’t listen to the audio. '
+              }
+              <Link
+                href="/demo"
+                size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.X_SMALL}
+              >
+                {'Learn more'}
+              </Link>
+            </Text>
+          }
+          cta={
+            <Button size={BUTTON_SIZES.SMALL} type={BUTTON_TYPES.SECONDARY}>
+              {'Replace File'}
+            </Button>
+          }
+          thumbnailCount={1}
+          thumbnail={
+            <Thumbnail
+              onClick={() => action('thumbnail action found')()}
+              type={THUMBNAIL_TYPES.IMAGE}
+              displayBackground={THUMBNAIL_BG[THUMBNAIL_TYPES.IMAGE]}
+              aria-label="my helper text describing this thumbnail image"
+            />
```

Comment:
I wonder if for these thumbnails we could just get them to take an `element` or `page` object plucked directly off the story, then just infer all of these props off of that object. Thoughts?

Reply:
I thought consistency with the layers panel would be useful here so we don't have to really do much reinventing the wheel. So for elements we'd just need to pull in `LayerIcon` from `getDefinition` ( panels/design/layer) or for a page, mimic the page carousel with `PagePreview`. The thumbnail stories are just using output from these.